py3status/modules/diskdata.py

```diff
@@ -87,17 +105,73 @@ def post_config_hook(self):
             value - value (float)
             unit - unit (string)
         """
-        self.last_diskstats = self._get_diskstats(self.disk)
+        types_info = {
+            "disks": {"major": [3, 8, 22]},
+            "raid": {"major": [9]},
+            "cdrom": {
+                "major": [
+                    11,
+                    15,
+                    16,
+                    17,
+                    18,
+                    20,
+                    23,
+                    24,
+                    25,
+                    26,
+                    27,
+                    28,
+                    29,
+                    30,
+                    32,
+                    33,
+                    34,
+                ]
+            },
+            "mmc": {"major": [31]},
+        }
+
+        self._disks = {}
+
+        if not self.disk:
+            allowed_types = []
+            for _type in self.types.split():
+                allowed_types.extend(types_info[_type]["major"])
+            with open("/proc/diskstats") as ds:
+                for stat in ds:
+                    line = stat.split()
+                    if int(line[0]) in allowed_types and int(line[1]) == 0:
+                        self._add_monitored_disk("/dev/" + line[2])
+        else:
+            for disk in self.disk.split():
+                # mountpoint or fully qualified device path
+                if disk.startswith("/"):
+                    # mountpoint
+                    if not disk.startswith("/dev/"):
+                        with open("/proc/mounts") as mounts:
+                            for entry in mounts:
```

Comment:
Knowing users won't specify full dev path, it may be better to open `proc/mounts` once whenever users specified anything or not... and reuse it.

Reply:
I do not get the point, proc/mounts here is used only when coping w/ mountpoints. Context manager could indeed be moved outside the loop with a seek(0) at every cycle, is that what you intend?

Reply:
Yes. Something like this. `/proc/mounts` looks ridiculous. 55 lines. 
```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index fdc024f..e806238 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -153,19 +153,19 @@ class Py3status:
                     if int(line[0]) in allowed_types and int(line[1]) == 0:
                         self._add_monitored_disk("/dev/" + line[2])
         else:
+            with open("/proc/mounts") as f:
+                mounts = [x.split() for x in reversed(f.readlines())]
             for disk in self.disks:
                 # mountpoint or fully qualified device path
                 if disk.startswith("/"):
                     # mountpoint
                     if not disk.startswith("/dev/"):
-                        with open("/proc/mounts") as mounts:
-                            for entry in mounts:
-                                mount = entry.split()
-                                if mount[1] == disk:
-                                    self._add_monitored_disk(mount[0])
+                        for mount in mounts:
+                            if mount[1] == disk:
+                                self._add_monitored_disk(mount[0])
+                                break
                     else:
                         self._add_monitored_disk(disk)
-
                 # non fully qualified device path
                 else:
                     self._add_monitored_disk("/dev/" + disk)
```

Reply:
did that